### PR TITLE
Migrate static data from hiera to puppet code

### PR DESCRIPTION
**Pull request by bastelfreak**
This allows us to display the hiera data in the REFERENCE.md and also
helps us to keep the defaults.yaml shorter.

### CODE REVIEW

**Comment by bastelfreak** [comment]
For reviewing I suggest https://github.com/voxpupuli/puppet-prometheus/pull/407/files?utf8=%E2%9C%93&diff=split&w=1

File: manifests/bird_exporter.pp [hunk 1]
```diff
@@ -64,9 +64,9 @@
   Boolean $purge_config_dir         = true,
   Boolean $restart_on_change        = true,
   Boolean $service_enable           = true,
-  String $service_ensure            = 'running',
-  String $service_name              = 'bird_exporter',
-  Optional[String] $init_style      = $prometheus::init_style,
+  String[1] $service_ensure         = 'running',
+  String[1] $service_name           = 'bird_exporter',
+  Prometheus::Initstyle $init_style     = $facts['service_provider'],
```
Comment (ekohl) [inline manifests/bird_exporter.pp]: Odd alignment?

File: manifests/alertmanager.pp [hunk 2]
```diff
@@ -107,23 +107,23 @@
...
-  Boolean $manage_service        = true,
-  Boolean $manage_user           = true,
-  String $os                     = $prometheus::os,
-  String $extra_options          = '',
-  Optional[String] $download_url = undef,
-  String $config_mode            = $prometheus::config_mode,
-  String $arch                   = $prometheus::real_arch,
-  Stdlib::Absolutepath $bin_dir  = $prometheus::bin_dir,
+  Boolean $service_enable           = true,
+  String[1] $service_ensure         = 'running',
```
Comment (ekohl) [inline manifests/alertmanager.pp:111]: Unrelated so probably better for a different PR, but there's a `Stdlib::Ensure::Service` type

File: manifests/daemon.pp [hunk 3]
```diff
@@ -61,7 +61,7 @@
   Boolean $purge                       = true,
   String $options                      = '',
   Prometheus::Initstyle $init_style    = $prometheus::init_style,
```
Comment (ekohl) [inline manifests/daemon.pp]:
```suggestion
  Prometheus::Initstyle $init_style    = $facts['service_provider'],
```

File: manifests/bird_exporter.pp (hunk 1 above)
Comment (bastelfreak) [reply manifests/bird_exporter.pp]: fixed it

File: manifests/alertmanager.pp (hunk 2 above)
Comment (bastelfreak) [reply manifests/alertmanager.pp:111]: thanks, will do this in another PR

**Comment by bastelfreak** [comment]
@ekohl I pushed an update

**Review by ekohl — APPROVED** [review]

**Pull request merged**